jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/BaseImageParameters.java

```diff
@@ -93,6 +98,17 @@ public void setCredHelper(String credHelper) {
     this.credHelper = credHelper;
   }
 
+  @Input
+  @Nullable
+  @Optional
+  public Map<String, String> getEnvironment() {
+    return environment;
+  }
+
+  public void setEnvironment(Map<String, String> environment) {
+    this.environment = environment;
+  }
```

Comment:
The way we want is to set this through `jib.from.credHelper.environment` as I showed in https://github.com/GoogleContainerTools/jib/issues/2814#issuecomment-783645262, not through `jib.from.environment` as you did here.

An important thing is to maintain backward compatibility. That is, all of these should work:
```
jib.from {
    image = ...
    credHelper = 'ecr-login`
}
```
```
jib.from {
    image = ...
    credHelper.helper = 'ecr-login'
}
```
```
jib.from {
    image = ...
    credHelper {
        helper = 'ecr-login'
        environment = ...
    }
}
```